programs/virtual-curve/src/instructions/migration/dynamic_amm_v2/migrate_damm_v2_initialize_pool.rs

```diff
@@ -379,16 +379,14 @@ pub fn handle_migrate_damm_v2<'c: 'info, 'info>(
         ctx.accounts.validate_config_key(&damm_config)?;
     }
 
-    let mut migration_metadata = ctx.accounts.migration_metadata.load_mut()?;
-    let migration_progress = MeteoraDammV2MetadataProgress::try_from(migration_metadata.progress)
-        .map_err(|_| PoolError::TypeCastFailed)?;
+    let mut virtual_pool = ctx.accounts.virtual_pool.load_mut()?;
 
     require!(
-        migration_progress == MeteoraDammV2MetadataProgress::Init,
+        virtual_pool.get_migration_progress()? == MigrationProgress::LockedVesting,
```

Comment:
Doesn't this break current pools that's eligible for migration but not yet migrated?

Reply:
Maybe we can support backward compatibility by
```
if virtual_pool.finish_curve_timestamp > 0 {
    require!(
        virtual_pool.get_migration_progress()? == MigrationProgress::LockedVesting,
        PoolError::NotPermitToDoThisAction
    );
}
else {
    require!(
        migration_progress == MeteoraDammV2MetadataProgress::Init,
        PoolError::NotPermitToDoThisAction
    );
}
```

Reply:
We are fine with breaking changes (keep to code clean)